RecoPixelVertexing/PixelTriplets/plugins/GPUCACell.h

```diff
@@ -288,50 +289,62 @@ class GPUCACell {
     // the ntuplets is then saved if the number of hits it contains is greater
     // than a threshold
 
-    auto doubletId = this - cells;
-    tmpNtuplet.push_back_unsafe(doubletId);
-    assert(tmpNtuplet.size() <= 5);
-
-    bool last = true;
-    for (unsigned int otherCell : outerNeighbors()) {
-      if (cells[otherCell].isKilled())
-        continue;  // killed by earlyFishbone
-      last = false;
-      cells[otherCell].find_ntuplets<DEPTH - 1>(
-          hh, cells, cellTracks, foundNtuplets, apc, quality, tmpNtuplet, minHitsPerNtuplet, startAt0);
-    }
-    if (last) {  // if long enough save...
-      if ((unsigned int)(tmpNtuplet.size()) >= minHitsPerNtuplet - 1) {
+    if constexpr (DEPTH <= 0) {
+      printf("ERROR: GPUCACellT::find_ntuplets reached full depth!\n");
+#ifdef __CUDA_ARCH__
+      __trap();
+#else
+      abort();
+#endif
+    } else {
+      auto doubletId = this - cells;
+      tmpNtuplet.push_back_unsafe(doubletId);
+      assert(tmpNtuplet.size() <=
+             int(TrackerTraits::maxHitsOnTrack -
+                 3));  //1 for the container, 1 because these are doublets, 1 because we may push another
```

Comment:
Yes, I moved it to `maxHitsOnTrack-1` because at the end of the chain there may be tuplets for which the two additional hits from the fishbone were not added and that are actually as long as the "full" tracker in the endcap that were wrongly triggering the assert while being short enough to be accommodated in the container

Reply:
Most probably I could rephrase the comment to be clearer.